internal/clientio/resp.go

```diff
@@ -152,17 +152,42 @@ func encodeString(v string) []byte {
 	return []byte(fmt.Sprintf("$%d\r\n%s\r\n", len(v), v))
 }
 
+// encodeBool encodes bool as simple strings
+func encodeBool(v bool) []byte {
+	var val string
+	switch v {
+	case true:
+		val = "true"
+	case false:
+		val = "false"
+	}
+	return []byte(fmt.Sprintf("+%v\r\n", val))
+}
+
 func Encode(value interface{}, isSimple bool) []byte {
 	switch v := value.(type) {
 	case string:
-		if isSimple {
+		// encode as simple strings
+		if isSimple || v == "[" || v == "{" {
 			return []byte(fmt.Sprintf("+%s\r\n", v))
 		}
+		// encode as bulk strings
 		return encodeString(v)
 	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
 		return []byte(fmt.Sprintf(":%d\r\n", v))
 	case float32, float64:
-		return []byte(fmt.Sprintf(":%v\r\n", v))
+		// since encoding/json unmarshals numeric values as floats
+		// we need to check if value is an integer
+		intValue, isInteger := utils.IsFloatToIntPossible(v.(float64))
+		if isInteger {
+			return []byte(fmt.Sprintf(":%d\r\n", intValue))
+		}
+
+		// if it is a float, encode like a string
+		str := strconv.FormatFloat(v.(float64), 'f', -1, 64)
+		return encodeString(str)
```

Comment:
Are we sure this modified logic doesn't end up messing up cases where someone stores a whole number in floating point representation? Something like `5.0`? How does redis behave in such cases?

Reply:
A whole number in floating point representation is shown as string on Redis and Dice master branch. 

On Dice master branch:
```
set a 1
OK
get a
(integer) 1
set a 1.34
OK
get a
"1.34"
set a 1.00
OK
get a
"1.00"
```
The behaviour of JSONResp feature branch is identical as master branch for above cases.

On Redis:
```
set a 1
OK
get a
"1" ---- Integers are also encoded as strings in Redis (different behaviour)
set a 1.34
OK
get a
"1.34"
set a 1.00
OK
get a
"1.00"
```

I am also working on #69. Let me take up integer difference there.

I did run all unit tests for eval_test.go and all integration tests and they all passed.